src/lib.rs

```diff
@@ -209,11 +209,50 @@ pub fn verify_signature<S: Into<String>>(_signature: S, _data: S, _pub_key: S) -
     Err(ErrorCode::FunctionNotImplemented)
 }
 
+#[derive(Serialize, Default)]
+struct CommitInputStruct {
```

Comment:
How about putting the structs inside `commit_entry()` since its the only place it is being used?

Reply:
There is only one way to find out... :)